src/client/datascience/data-viewing/jupyterVariableDataProvider.ts

```diff
@@ -53,6 +54,21 @@ export class JupyterVariableDataProvider implements IJupyterVariableDataProvider
         });
     }
 
+    // Parse a string of the form (1, 2, 3)
+    private static parseShape(shape: string) {
+        try {
+            if (shape.startsWith('(') && shape.endsWith(')')) {
+                return shape
+                    .substring(1, shape.length - 1)
+                    .split(',')
+                    .map((shapeEl) => parseInt(shapeEl));
+            }
+        } catch (e) {
+            traceError(`Could not parse IJupyterVariable with malformed shape: ${shape}`);
```

Comment:
Would be a good telemetry point to see how often this fails for users.
This way we'd know its not always working as expected & we need to look into this.

Reply:
This code in particular should never fail, because the `shape` attribute comes from the Python code and in fact we validate `shape` in the Python scripts as well. Not convinced this is worth sending new telemetry for.